library/src/main/java/com/getstream/sdk/chat/notifications/StreamNotificationsManager.java

```diff
@@ -0,0 +1,331 @@
+                    .setCategory(NotificationCompat.CATEGORY_MESSAGE)
+                    .setShowWhen(true)
+                    .setContentIntent(
+                            notificationOptions.getNotificationIntentProvider()
+                                    .getIntentForWebSocketEvent(context, event)
+                    );
+
+            String messageId = event.getMessage().getId();
+
+            if (lastNotificationId == null || !lastNotificationId.equals(messageId)) {
+                lastNotificationId = messageId;
+                showNotification(notificationId, builder.build(), context);
+            } else {
+                Log.i(TAG, "Notification with id:" + messageId + " already showed");
+            }
+        }
+    }*/
+
+    @Override
+    public void onReceiveWebSocketEvent(@NotNull Event event, @NotNull Context context) {
+        Log.d(TAG, "onReceiveWebSocketEvent: " + event);
+
+        handleEvent(context, event);
+    }
+
+    /**
+     * Handle WebSocket message, check and show notifications
+     *
+     * @param context - Context
+     * @param event   - event from websocket
+     */
+    public void handleEvent(Context context, Event event) {
+        if (event.getType() == EventType.MESSAGE_NEW) {
+
+            String messageId = event.getMessage().getId();
+
+            if (lastNotificationId == null || !lastNotificationId.equals(messageId)) {
+                lastNotificationId = messageId;
+                loadMessage(context, messageId);
+            } else {
+                Log.i(TAG, "Notification with id:" + messageId + " already showed");
+            }
+        }
+    }
+
+    /**
+     * Handle Firebase messageText, check and show notifications
+     *
+     * @param context       - Context
+     * @param remoteMessage - msg from Firebase
+     */
+    public void handleRemoteMessage(Context context, RemoteMessage remoteMessage) {
+        String messageId = remoteMessage.getData().get(FIREBASE_MESSAGE_ID_KEY);
+        this.remoteMessage = remoteMessage;
+
+        if (checkSentNotificationWithId(messageId)) {
+
+            lastNotificationId = messageId;
+            notificationId = (int) System.currentTimeMillis();
+
+            if (messageId != null && !messageId.isEmpty()) {
+                loadMessage(context, messageId);
+            } else {
+                Log.e(TAG, "RemoteMessage: messageId =" + messageId);
+            }
+        }
+    }
+
+    private void loadMessage(Context context, @NonNull String messageId) {
+        StreamChat.getInstance(context).getMessage(messageId, new MessageCallback() {
+            @Override
+            public void onSuccess(MessageResponse response) {
+                onMessageLoaded(context, response.getMessage());
+            }
+
+            @Override
+            public void onError(String errMsg, int errCode) {
+                Log.e(TAG, "Can\'t load user image. Error: " + errMsg);
```

Comment:
Currently implementation ignores errors completely, so if message is not loaded, then user will never see notification. Talked with Tommaso about error handling and we agreed to show generic notification where title is app name and message is just something like `You've received new message`. And there should be setter for calback which developers can provide and show their own notification, but not generic one.

Reply:
@elevenetc Just in case. So, should i remove loading for message and show only generic notification where title is app name and message or i should to show generic message for error loading only?
Because for now i understand it like remove method loading message (and all logic with actions) and show default notification ("You've received new message").

Reply:
Keep loading message. But if messages is not loaded because of error, then show generic notification.

So:
1. notification received
2. load message
2.1 load success: show notification. End.
2.2 load failed
2.2.1 if developer provided failed callback, then call it
2.2.2 if developer didn't provide failed callback, then show notification